core/templates/pages/blog-post-page/blog-post-page.component.ts

```diff
@@ -93,11 +93,11 @@ export class BlogPostPageComponent implements OnInit {
     selection?.removeAllRanges();
   }
 
-  decodeAuthorProfilePicUrl(url: string): void {
-    this.DEFAULT_PROFILE_PICTURE_URL = this.urlInterpolationService
-      .getStaticImageUrl('/general/no_profile_picture.png');
-    this.authorProfilePicUrl = decodeURIComponent((
-      url || this.DEFAULT_PROFILE_PICTURE_URL));
+  getAuthorProfilePicUrl(username: string): void {
+    let profileImageUrls = this.userService.getProfileImageDataUrlAsync(
+      username);
+    this.authorProfilePicPngUrl = profileImageUrls[0];
+    this.authorProfilePicWebpUrl = profileImageUrls[1];
```

Comment:
Should this be called from the HTML

Reply:
I have updated all the parts where we had an extra function for setting the profile pictures.